include/librealsense2/h/rs_device.h

```diff
@@ -353,6 +353,151 @@ rs2_raw_data_buffer* rs2_serialize_json(rs2_device* dev, rs2_error** error);
 /* Load JSON and apply advanced-mode controls */
 void rs2_load_json(rs2_device* dev, const void* json_content, unsigned content_size, rs2_error** error);
 
+/**
+* \brief Creates RealSense firmware log message.
+* \param[in] dev	        Device from which the FW log will be taken using the created message
+* \param[out] error         If non-null, receives any error that occurs during this call, otherwise, errors are ignored.
+* \return                   pointer to created empty firmware log message
+*/
+rs2_firmware_log_message* rs2_create_firmware_log_message(rs2_device* dev, rs2_error** error);
+
+/**
+* \brief Gets RealSense firmware log.
+* \param[in] dev	        Device from which the FW log should be taken
+* \param[in] fw_log_msg	    Firmware log message object to be filled
+* \param[out] error         If non-null, receives any error that occurs during this call, otherwise, errors are ignored.
+* \return                   true for success, false for failure - failure happens if no firmware log was sent by the hardware monitor
+*/
+int rs2_get_firmware_log(rs2_device* dev, rs2_firmware_log_message** fw_log_msg, rs2_error** error);
+
+int rs2_get_number_of_flash_logs(rs2_device* dev, rs2_error** error);
```

Comment:
I think you don't need it
Lets have it operate like firmware logs - it returns log messages until there are no more less